bokeh/io/saving.py

```diff
@@ -96,7 +96,7 @@ def save(obj: LayoutDOM, filename: PathLike | None = None, resources: ResourcesL
 
     filename, resources, title = _get_save_args(state, filename, resources, title)
     _save_helper(obj, filename, resources, title, template, theme)
-    return abspath(filename)
+    return abspath(expanduser(filename))
```

Comment:
The same thing as `export_png`, `open` inside `_save_helper` does not work with `~`. 

I can remove the `expanduser` if this is the desired behavior.

``` python
import os

from bokeh.plotting import figure, save

p = figure(x_range=(0, 100), y_range=(0, 100), toolbar_location=None)

filename = save(p, filename=os.path.expanduser("~/index.html"))  # Works
filename = save(p, filename="~/index.html")  # Raises FileNotFoundError

```

``` python-traceback
Traceback (most recent call last):
  File "/home/shh/folder with space/save2.py", line 8, in <module>
    filename = save(p, filename="~/index.html")  # Raises FileNotFoundError
  File "/home/shh/Downloads/bokeh/bokeh/io/saving.py", line 98, in save
    _save_helper(obj, filename, resources, title, template, theme)
  File "/home/shh/Downloads/bokeh/bokeh/io/saving.py", line 166, in _save_helper
    with open(filename, mode="w", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: '~/index.html'
```